### PR TITLE
Add rsw

### PR DESCRIPTION
**Pull request by totto82**
Ported from OPM/opm-material#543

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Review by atgeirr — COMMENTED** [review]
I think this is good, except where noted in the comments below.

File: opm/material/fluidsystems/BlackOilFluidSystem.hpp [hunk 1]
```diff
@@ -968,12 +1039,12 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
 
         const auto& p = decay<LhsEval>(fluidState.pressure(phaseIdx));
         const auto& T = decay<LhsEval>(fluidState.temperature(phaseIdx));
-        const auto& saltConcentration = decay<LhsEval>(fluidState.saltConcentration());
+        const auto& saltConcentration = BlackOil::template getSaltConcentration_<ThisType, FluidState, LhsEval>(fluidState, regionIdx);
```
[the remark this answers is not in the document]
Comment (totto82) [reply opm/material/fluidsystems/BlackOilFluidSystem.hpp:1042]: The new code will allow for a fluidstate without  a method called saltConcentration(). There are still some call left to fluidState.saltConcentration(), I only "fixed" the part of the code I was changing anyway. This also align more with how we do it for rs and rv etc.

File: opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.hpp [hunk 2]
```diff
@@ -211,6 +223,33 @@ class BrineCo2Pvt
...
+     */
+    template <class Evaluation>
+    Evaluation viscosity(unsigned regionIdx,
+                         const Evaluation& temperature,
+                         const Evaluation& pressure,
+                         const Evaluation& /*Rsw*/,
+                         const Evaluation& /*saltConcentration*/) const
+    {
+        //TODO: The viscosity does not yet depend on the composition
+        return saturatedViscosity(regionIdx, temperature, pressure);
```
[the remark this answers is not in the document]
Comment (totto82) [reply opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.hpp:250]: One candidate is to use the LBC model for co2rich mixture (that we already have in opm/material/viscocitymodels)

File: opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.hpp [hunk 3]
```diff
@@ -222,6 +261,30 @@ class BrineCo2Pvt
...
+     * \brief Returns the formation volume factor [-] of the fluid phase.
+     */
+    template <class Evaluation>
+    Evaluation inverseFormationVolumeFactor(unsigned regionIdx,
+                                            const Evaluation& temperature,
+                                            const Evaluation& pressure,
+                                            const Evaluation& Rs,
+                                            const Evaluation& /*saltConcentration*/) const
+    {
+        return inverseFormationVolumeFactor(regionIdx, temperature, pressure, Rs);
```
[the remark this answers is not in the document]
Comment (totto82) [reply opm/material/fluidsystems/blackoilpvt/BrineCo2Pvt.hpp:286]: No it is not introduced here. I think a good solution would be to stop using the pvt models for the phases directly (they should be internal to the FluidSystem) and instead call FluidSystem::density(fs, ...) etc. that would avoid the potential for misuse.

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

File: opm/material/fluidsystems/BlackOilFluidSystem.hpp [hunk 4]
```diff
@@ -769,9 +815,10 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
         }
 
         case waterPhaseIdx:
+            const LhsEval Rsw(0.0);
```
[the remark this answers is not in the document]
Comment (atgeirr) [reply opm/material/fluidsystems/BlackOilFluidSystem.hpp:826]: By brine I mean with salt, but now I do not see why I thought something was missing, it is taken care of with saltConcentration as evaluated on line 756 it seems.

**Review by atgeirr — APPROVED** [review]

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
jenkins build this opm-models=763 opm-simulators=4292 please

**Comment by totto82** [comment]
All is green and all is approved. Will merge! Thanks @atgeirr for the review.

**Pull request merged**